File: docs/vsto/codesnippet/VisualBasic/Trin_VstcoreActionsPaneWordVB/ThisDocument.vb
Public Class ThisDocument

    '--------------------------------------------------------------------------
    '<Snippet10>
    Dim insertText As New InsertTextControl
    '</Snippet10>

    '<Snippet6>
    Dim actions As New ActionsControl
    '</Snippet6>

    '<Snippet14>
    Dim hello As New HelloControl
    '</Snippet14>

    '<Snippet16>
    Dim addText As New AddTextControl
    Dim showProperties As New ShowPropertiesControl
    Dim showAll As New ShowAllControl

    Friend WithEvents showCheck As System.Windows.Forms.CheckBox
    '</Snippet16>


    '--------------------------------------------------------------------------
    Sub StartUp1()

        '<Snippet7>
        Me.ActionsPane.Controls.Add(actions)
        '</Snippet7>


        '<Snippet11>
        Me.ActionsPane.Controls.Add(insertText)
        '</Snippet11>


        '<Snippet15>
        Me.ActionsPane.Controls.Add(hello)
        '</Snippet15>


        '<Snippet17>
        Me.ActionsPane.Controls.Add(showAll)
        showCheck = showAll.showCheck
        '</Snippet17>

        '<Snippet34>
        Me.Application.TaskPanes(Word.WdTaskPanes.wdTaskPaneDocumentActions).Visible = False
        '</Snippet34>
    End Sub


    '--------------------------------------------------------------------------
    '<Snippet18>
    Private Sub SampleInsertNode_ContextEnter(ByVal sender As Object, _
        ByVal e As Microsoft.Office.Tools.Word.ContextChangeEventArgs) _
        Handles SampleInsertNode.ContextEnter

        If showAll.showCheck.Checked = False Then
            Me.ActionsPane.Controls.Add(addText)
            Me.ActionsPane.Controls.Remove(showProperties)
        End If
    End Sub
    '</Snippet18>


    '<Snippet19>
    Private Sub SampleInsertNode_ContextLeave(ByVal sender As Object, _
        ByVal e As Microsoft.Office.Tools.Word.ContextChangeEventArgs) _
        Handles SampleInsertNode.ContextLeave

        If showAll.showCheck.Checked = False Then
            Me.ActionsPane.Controls.Remove(addText)
        End If
    End Sub
    '</Snippet19>


    '<Snippet20>
    Private Sub SampleTableNode_ContextEnter(ByVal sender As Object, _
        ByVal e As Microsoft.Office.Tools.Word.ContextChangeEventArgs) _
        Handles SampleTableNode.ContextEnter

        If showAll.showCheck.Checked = False Then
            Me.ActionsPane.Controls.Remove(addText)
            Me.ActionsPane.Controls.Add(showProperties)
        End If

        showProperties.tableProperties.Enabled = True
    End Sub
    '</Snippet20>


    '<Snippet21>
    Private Sub SampleTableNode_ContextLeave(ByVal sender As Object, _
        ByVal e As Microsoft.Office.Tools.Word.ContextChangeEventArgs) _
        Handles SampleTableNode.ContextLeave

        If showAll.showCheck.Checked = False Then
            Me.ActionsPane.Controls.Remove(showProperties)
        End If

        showProperties.tableProperties.Enabled = False
    End Sub
    '</Snippet21>


    '--------------------------------------------------------------------------
    '<Snippet23>
    Private Sub showCheck_CheckStateChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) _
        Handles showCheck.CheckStateChanged

        If showAll.showCheck.Checked = True Then
            Me.ActionsPane.Controls.Add(addText)
            Me.ActionsPane.Controls.Add(showProperties)
        Else
            Me.ActionsPane.Controls.Remove(addText)
            Me.ActionsPane.Controls.Remove(showProperties)
        End If
    End Sub
    '</Snippet23>


    '--------------------------------------------------------------------------
    '<Snippet29>
    '<Snippet30>
    Private Sub ActionsPane_OrientationChanged(ByVal sender As Object, ByVal e As System.EventArgs) _
        Handles ActionsPane.OrientationChanged

        If Me.ActionsPane.Orientation = Orientation.Horizontal Then
    '</Snippet30>

    '<Snippet31>
            Me.ActionsPane.StackOrder = Microsoft.Office.Tools.StackStyle.FromLeft
        Else
            Me.ActionsPane.StackOrder = Microsoft.Office.Tools.StackStyle.FromTop
        End If
    End Sub
    '</Snippet31>
    '</Snippet29>

    Sub Position()
        '<Snippet100>
        Me.CommandBars("Task Pane").Position = _
            Microsoft.Office.Core.MsoBarPosition.msoBarLeft
        '</Snippet100>
    End Sub

    Sub SetTopandLeft()
        '<Snippet101>
        If Me.CommandBars("Task Pane").Position = _
            Microsoft.Office.Core.MsoBarPosition.msoBarFloating Then

            Me.CommandBars("Task Pane").Top = 0
            Me.CommandBars("Task Pane").Left = 0

        End If
        '</Snippet101>
    End Sub

    Sub SetAPWidth()
        '<Snippet102>
        Me.CommandBars("Task Pane").Width = 200
        '</Snippet102>
    End Sub

End Class

File: docs/vsto/codesnippet/CSharp/Trin_VstcoreActionsPaneWordCS/ThisDocument.cs
using System;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using Microsoft.VisualStudio.Tools.Applications.Runtime;
using Word=Microsoft.Office.Interop.Word;
using Office=Microsoft.Office.Core;

namespace Trin_VstcoreActionsPaneWordCS
{
    public partial class ThisDocument
    {
        //---------------------------------------------------------------------
        //<Snippet10>
        private InsertTextControl insertText = new InsertTextControl();
        //</Snippet10>

        //<Snippet6>
        private ActionsControl actions = new ActionsControl();
        //</Snippet6>

        //<Snippet14>
        private HelloControl hello = new HelloControl();
        //</Snippet14>

        //<Snippet16>
        private AddTextControl addText = new AddTextControl();
        private ShowPropertiesControl showProperties = new ShowPropertiesControl();
        private ShowAllControl showAll = new ShowAllControl();
        //</Snippet16>


        //---------------------------------------------------------------------
        private void StartUp(object sender, System.EventArgs e)
        {
            //<Snippet7>
            this.ActionsPane.Controls.Add(actions);
            //</Snippet7>


            //<Snippet11>
            this.ActionsPane.Controls.Add(insertText);
            //</Snippet11>


            //<Snippet15>
            this.ActionsPane.Controls.Add(hello);
            //</Snippet15>


            //<Snippet17>
            this.ActionsPane.Controls.Add(showAll);
            //</Snippet17>

            //<Snippet34>
            this.Application.TaskPanes[Word.WdTaskPanes.wdTaskPaneDocumentActions].Visible = false;
            //</Snippet34>


            //<Snippet22>
            this.SampleInsertNode.ContextEnter +=new Microsoft.Office.Tools.Word.
                ContextChangeEventHandler(SampleInsertNode_ContextEnter); 
            
            this.SampleInsertNode.ContextLeave += new Microsoft.Office.Tools.Word.
                ContextChangeEventHandler(SampleInsertNode_ContextLeave);

            this.SampleTableNode.ContextEnter += new Microsoft.Office.Tools.Word.
                ContextChangeEventHandler(SampleTableNode_ContextEnter);

            this.SampleTableNode.ContextLeave += new Microsoft.Office.Tools.Word.
                ContextChangeEventHandler(SampleTableNode_ContextLeave);
            //</Snippet22>


            //<Snippet24>
            showAll.showCheck.CheckStateChanged += new EventHandler(showCheck_CheckStateChanged);
            //</Snippet24>
        }



        //---------------------------------------------------------------------
        private void InternalStartup()
        {
            this.Startup += new System.EventHandler(ThisDocument_Startup);
            this.Startup += new System.EventHandler(StartUp);
        }


        //---------------------------------------------------------------------
        //<Snippet18>
        private void SampleInsertNode_ContextEnter(object sender,
            Microsoft.Office.Tools.Word.ContextChangeEventArgs e)
        {
            if (showAll.showCheck.Checked == false)
            {
                this.ActionsPane.Controls.Add(addText);
                this.ActionsPane.Controls.Remove(showProperties);
            }
        }
        //</Snippet18>


        //<Snippet19>
        private void SampleInsertNode_ContextLeave(object sender,
            Microsoft.Office.Tools.Word.ContextChangeEventArgs e)
        {
            if (showAll.showCheck.Checked == false)
            {
                this.ActionsPane.Controls.Remove(addText);
            }
        }
        //</Snippet19>


        //<Snippet20>
        private void SampleTableNode_ContextEnter(object sender,
            Microsoft.Office.Tools.Word.ContextChangeEventArgs e)
        {
            if (showAll.showCheck.Checked == false)
            {
                this.ActionsPane.Controls.Remove(addText);
                this.ActionsPane.Controls.Add(showProperties);
            }
            showProperties.tableProperties.Enabled = true;
        }
        //</Snippet20>


        //<Snippet21>
        private void SampleTableNode_ContextLeave(object sender,
            Microsoft.Office.Tools.Word.ContextChangeEventArgs e)
        {
            if (showAll.showCheck.Checked == false)
            {
                this.ActionsPane.Controls.Remove(showProperties);
            }
            showProperties.tableProperties.Enabled = false;
        }
        //</Snippet21>


        //---------------------------------------------------------------------
        //<Snippet23>
        private void showCheck_CheckStateChanged(object sender, EventArgs e)
        {
            if (showAll.showCheck.Checked)
            {
                this.ActionsPane.Controls.Add(addText);
                this.ActionsPane.Controls.Add(showProperties);
            }
            else
            {
                this.ActionsPane.Controls.Remove(addText);
                this.ActionsPane.Controls.Remove(showProperties);
            }
        }
        //</Snippet23>


        //---------------------------------------------------------------------
        //<Snippet29>
        //<Snippet32>
        private void ThisDocument_Startup(object sender, System.EventArgs e)
        {
            this.ActionsPane.OrientationChanged += new EventHandler(ActionsPane_OrientationChanged);
        }
        //</Snippet32>

        //<Snippet30>
        private void ActionsPane_OrientationChanged(object sender, EventArgs e)
        {
            if (ActionsPane.Orientation == Orientation.Horizontal)
            {
        //</Snippet30>
        //<Snippet31>
                this.ActionsPane.StackOrder = Microsoft.Office.Tools.StackStyle.FromLeft;
            }
            else
            {
                this.ActionsPane.StackOrder = Microsoft.Office.Tools.StackStyle.FromTop;
            }
        }
        //</Snippet31>
        //</Snippet29>

        private void Position()
        {
            //<Snippet100>
            this.CommandBars["Task Pane"].Position =
                Microsoft.Office.Core.MsoBarPosition.msoBarLeft;
            //</Snippet100>
        }

        private void SetTopandLeft()
        {
        //<Snippet101>
            if (this.CommandBars["Task Pane"].Position == 
                Microsoft.Office.Core.MsoBarPosition.msoBarFloating)
            {
                this.CommandBars["Task Pane"].Top = 0;
                this.CommandBars["Task Pane"].Left = 0;
            }
        //</Snippet101>
        }

       private void SetAPWidth()
       {
           //<Snippet102>
           this.CommandBars["Task Pane"].Width = 200;
           //</Snippet102>
       }
     }
}
